types/src/database_sql.rs

```diff
@@ -398,7 +396,76 @@ impl Database {
             "queries/update_word.sql",
             word.id,
             &source as _,
-            &commentary as _
+            &commentary as _,
+        )
+        .execute(&self.client)
+        .await?;
+
+        // If word segmentation was not changed, then return early since SQL update queries need to be called.
+        if word.segments.is_undefined() {
+            return Ok(word.id);
+        }
+
+        let document_id = query_file!("queries/word_by_id.sql", word.id)
```

Comment:
Couldn't you get this info from the update word query or something? Seems excessive here to have an extra query just to get the document ID, no?